packages/flutter/test/rendering/layers_test.dart

```diff
@@ -465,4 +465,213 @@ void main() {
+        expect(absorbed, isFalse);
+        expect(pathToValues(result.path), <int>[21, 11, 1, 0]);
+      });
+
+      test('when opaque should stop checking following children if one child absorbs, and return true', () {
+
+        // Tests:
+        //
+        //     o            (child3, translucent)
+        //     | x          (child2, opaque)
+        //     | | o        (child1)
+        //      \|/
+        //       x          (root, opaque)
+
+        final ContainerLayer root = AnnotatedRegionLayer<int>(0, opaque: true);
+        final ContainerLayer child1 = AnnotatedRegionLayer<int>(1);
+        root.append(child1);
+        final ContainerLayer child2 = AnnotatedRegionLayer<int>(2, opaque: true);
+        root.append(child2);
+        final ContainerLayer child3 = AnnotatedRegionLayer<int>(3);
+        root.append(child3);
+
+        final LayerHitTestResult<int> result = LayerHitTestResult<int>();
+        final bool absorbed = root.hitTest(result, regionOffset: Offset.zero);
+        expect(absorbed, isTrue);
+        expect(pathToValues(result.path), <int>[3, 2, 0]);
+      });
+
+      test('when translucent should stop checking following children if one child absorbs, and return false', () {
+
+        // Tests:
+        //
+        //     o            (child3, translucent)
+        //     | x          (child2, opaque)
+        //     | | o        (child1)
+        //      \|/
+        //       o          (root, translucent)
+
+        final ContainerLayer root = AnnotatedRegionLayer<int>(0);
+        final ContainerLayer child1 = AnnotatedRegionLayer<int>(1);
+        root.append(child1);
+        final ContainerLayer child2 = AnnotatedRegionLayer<int>(2, opaque: true);
+        root.append(child2);
+        final ContainerLayer child3 = AnnotatedRegionLayer<int>(3);
+        root.append(child3);
+
+        final LayerHitTestResult<int> result = LayerHitTestResult<int>();
+        final bool absorbed = root.hitTest(result, regionOffset: Offset.zero);
+        expect(absorbed, isFalse);
+        expect(pathToValues(result.path), <int>[3, 2, 0]);
+      });
+
+      test('should not add to opacity if the layer does not contains', () {
+
+        // Tests:
+        //         v
+        //    ──────────        (child1, translucent)
+        //     │
+        //    ────              (root, opaque)
+
+        const Offset position = Offset(50, 5);
+
+        final ContainerLayer root = AnnotatedRegionLayer<int>(
+          0, size: const Size(20, 10), opaque: true);
+        final ContainerLayer child1 = AnnotatedRegionLayer<int>(
+          1, size: const Size(100, 10), opaque: false);
+        root.append(child1);
+
+        final LayerHitTestResult<int> result = LayerHitTestResult<int>();
+        final bool absorbed = root.hitTest(result, regionOffset: position);
+        expect(absorbed, isFalse);
+        expect(pathToValues(result.path), <int>[1]);
+      });
+    });
+  });
```

Comment:
minor thing, but I would encourage you to not use `group` unless you have to. We've ended up with annoying bugs when we start using `group`s in more complicated ways, so we're generally moving away from them.